core/src/impl/Kokkos_Half_MathematicalFunctions.hpp

```diff
@@ -228,6 +229,92 @@ KOKKOS_INLINE_FUNCTION bool isnan(Kokkos::Experimental::bhalf_t x) {
       ((bit_pattern_x.value & fraction_mask.value) != 0));
 }
 #endif
+
+template <typename fp16_t>
+KOKKOS_INLINE_FUNCTION fp16_t nextafter_impl(fp16_t from, fp16_t to) {
```

Comment:
Let's rename to `nextafter_half_impl` since the name is a bit generic and we then static assert anyway.